cmd/ndm_daemonset/controller/controller.go

```diff
@@ -183,17 +187,34 @@ func (c *Controller) newClientSet() (client.Client, error) {
 	return clientSet, nil
 }
 
-// setNodeName set HostName field in Controller struct
-// if it gets from env else it returns error
-func (c *Controller) setNodeName() error {
-	host, ret := os.LookupEnv("NODE_NAME")
-	if ret != true {
-		return errors.New("error building hostname")
+// setHostName set HostName field in Controller struct
+// from the labels in node object
+func (c *Controller) setHostName() error {
+	nodeName, err := getNodeName()
+	if err != nil {
+		return fmt.Errorf("unable to get hostname: %v", err)
+	}
+	// get the node object and fetch the hostname label from the
+	// node object
+	node := &v1.Node{}
+	err = c.Clientset.Get(context.TODO(), client.ObjectKey{Namespace: "", Name: nodeName}, node)
+	if err != nil {
+		return err
 	}
-	c.HostName = host
+	c.HostName = node.Labels[NDMHostKey]
```

Comment:
n_: It is expected that this label will be always persent, but not guaranteed.. let us put an empty string check and fallback to nodename or error out here.